stubs/PyYAML/yaml/dumper.pyi

```diff
@@ -6,56 +8,56 @@ from yaml.serializer import Serializer
 class BaseDumper(Emitter, Serializer, BaseRepresenter, BaseResolver):
     def __init__(
         self,
-        stream,
-        default_style=...,
-        default_flow_style=...,
-        canonical=...,
-        indent=...,
-        width=...,
-        allow_unicode=...,
-        line_break=...,
-        encoding=...,
-        explicit_start=...,
-        explicit_end=...,
-        version=...,
-        tags=...,
+        stream: IO[str] | IO[bytes],
```

Comment:
Using`IO` is deprecated. Instead, this should use a protocol. `stream` is passed to `Emitter`, which has the following comment:

```
# The stream should have the methods `write` and possibly `flush`.
```

For now, the easiest way is to use `_typeshed.SupportsWrite[bytes | str]`, although ideally this would use overloads, depending on the `encoding` argument.

Same for the two methods below.